Common/Source/Comm/device.cpp

```diff
@@ -674,55 +693,79 @@ PDeviceDescriptor_t devGetDeviceOnPort(int Port){
 
  // devParseStream(devIdx, c, &GPS_INFO);
 BOOL devParseStream(int portNum, char* stream, int length, NMEA_INFO *pGPS){
-  
-  PDeviceDescriptor_t d = devGetDeviceOnPort(portNum);
-  if (d && d->ParseStream) {
-    d->HB=LKHearthBeats;
-    if (d->ParseStream(d, stream, length, pGPS)) {
-          
-    }
-    return(TRUE);        
-  }
+bool  ret = FALSE;
+PDeviceDescriptor_t din = devGetDeviceOnPort(portNum);
+PDeviceDescriptor_t d = NULL;
 
-  return(FALSE);    
+    for(int dev =0; dev < NUMDEV; dev++)
+    {
+      d = &DeviceList[dev];
+      if (din && d && d->ParseStream)
+      {
+       if((d->iSharedPort == portNum) ||  (d->Port == portNum))
+       {
+         d->HB=LKHearthBeats;
+         if (d->ParseStream(din, stream, length, pGPS)) {
+          
+         }
+         ret =TRUE;
+       }
```

Comment:
`d->ParseStream(d,` ....)` 
instead of  
`d->ParseStream(din, ...)`

and `ret =TRUE;` inside "if" condition